pkg/hub/controllers/slicegateway_controller.go

```diff
@@ -73,8 +74,12 @@ func (r *SliceGwReconciler) Reconcile(ctx context.Context, req reconcile.Request
 		log.Info("sliceGw doesn't belong to this cluster", "sliceGw", sliceGw.Name, "cluster", clusterName, "slicegw cluster", sliceGw.Spec.LocalGatewayConfig.ClusterName)
 		return reconcile.Result{}, nil
 	}
+	requeue, result, err := r.handleSliceGWDeletion(sliceGw, ctx, req)
```

Comment:
where are you adding this finalizer?
if you are not adding how will you get an event? 
PTAL whenever you get time

Reply:
Adding it inside this function on line number 213